propagators/opentelemetry_process_propagator/src/opentelemetry_process_propagator.erl

```diff
@@ -0,0 +1,66 @@
+-module(opentelemetry_process_propagator).
+
+-export([fetch_ctx/1,
+         fetch_parent_ctx/0,
+         fetch_parent_ctx/1,
+         fetch_parent_ctx/2]).
+
+-spec fetch_parent_ctx() -> opentelemetry:span_ctx() | undefined.
+fetch_parent_ctx() ->
+    fetch_parent_ctx(1, '$ancestors').
+
+-spec fetch_parent_ctx(non_neg_integer()) -> opentelemetry:span_ctx() | undefined.
+fetch_parent_ctx(MaxDepth) ->
```

Comment:
The Elixir API shows this as taking `key` and not depth. The example also passes a key to `fetch_parent_ctx/1` in the Elixir docs. See @dvic 's comment as well.

Reply:
I made it work in #49 , however, is it common to have `:undefined` returned in Elixir code? Shouldn't this just be `nil`?

Reply:
Don't get me started ;). The fact Elixir has `nil` is a big annoyance when doing libs you want used from both Erlang and Elixir.

Since this is Erlang code it is right to be `undefined`, technically the Elixir wrapper could convert it to `nil` but I'm not actually sure this application needs the Elixir wrapper that it comes with. I was going to bring that up with Bryan separately.

I'm fine with it having one, but the Otel API has an Elixir interface because it utilizes macros in both Erlang and Elixir and you can't use Erlang macros in Elixir, so it was required. Then the non-macro wrappers were added to make it feel consistent, since the user would otherwise be forced to use both wrapped and unwrapped which seemed annoying.

Reply:
I expect more pieces to this over time where macros would be added. There's also the side benefit of it being much easier to have legit docs. :/

Reply:
Ah ok.

But docs shouldn't be an issue `opentelemetry` and `opentelemetry_exporter` use `ex_doc` even though they are Erlang apps.

Reply:
Bah. I forgot to change the Elixir API.

Reply:
2cc: another idiomatic Elixir idiom is `{:ok, value} | :error`. `Map.fetch/2` does that. Does Erlang uses the same idiom? If it does, that may be an alternative that keep both consistent :)

Reply:
The reason to return `span_ctx() | undefined` instead of something like `{:ok, value} | :error` is that neither is an error and neither is more correct than the other.

`undefined` is accepted by any function that accepts a SpanCtx (which in most cases just results in nothing being done) because having no SpanCtx is a common case.

In OpenCensus we actually used specific types for this instead of `| undefined` all over, so would be like:

```
-type maybe_span_ctx() :: span_ctx() | undefined.
```

Then functions like this would return `maybe_span_ctx()`.